src/r_util.cpp

```diff
@@ -516,8 +516,9 @@ namespace rhost {
             std::vector<std::wstring> files;
             r_top_level_exec([&]() {
                 for (R_len_t i = 0; i < len; ++i) {
-                    const wchar_t* file_path = Rf_wtransChar(STRING_ELT(paths, i));
-                    files.emplace_back(file_path);
+                    const char* file_path_str = Rf_translateCharUTF8(STRING_ELT(paths, i));
+                    fs::path file_path(file_path_str);
```

Comment:
Don't go through the locale encoding though, because that would be potentially lossy (your codepage might not be able to represent an arbitrary Unicode pathname, but your filesystem might have files with such names).